### ISSUE TITLE
Chained array operations, in efficient calculation order

### ISSUE THREAD

**Issue by zbpetersbuf** [ubsuny/CP1-24-HW2#35]
I ll work on this task for the code

**Comment by iamtomato761** [comment]
great! I'll do the documentation

**Comment by zbpetersbuf** [comment]
what does this mean exactly? do you need anything from me to do this?

**Comment by laserlab** [comment]
@iamtomato761 @zbpetersbuf yes and no, you can add your code and ideally you work together ion the docstrings and @iamtomato761 can write a short description of the general idea of the function.

**Comment by zbpetersbuf** [comment]
are the doc strings the little boxes you add to Jupiter notebook?

**Comment by laserlab** [comment]
docstrings are python documentation, see for example here: https://peps.python.org/pep-0257/
or the example from class: https://github.com/ubsuny/CompPhys/blob/main/ReviewPython/docstrings/multiline.ipynb
